src/reducers/workspaces.ts

```diff
@@ -205,17 +202,16 @@ export const reducer: Reducer<IWorkspaceManagerState> = (
         ...state,
         [location]: {
           ...state[location],
-          output: newOutput,
-          isRunning: false
+          output: newOutput
         }
       }
-    case INTERRUPT_EXECUTION:
+    case END_INTERRUPT_EXECUTION:
+      /**
+       * Called to force a state change, which
+       * will cause a re-render
+       */
       return {
-        ...state,
-        [location]: {
-          ...state[location],
-          isRunning: false
-        }
+        ...state
```

Comment:
As discussed, there is a chance that this fails due to its async nature.

interrupt(context) in the saga inserts an interrupt error into the context. The scheduler only stops on the next evaluation step when it comes across this interrupt error. Then, the scheduler stops execution and sets context.isRunning to false. END_INTERRUPT_EXECUTION depends on the scheduler having done all this before the reducer takes this action of recreating the state, so that the props has a new value for isRunning and therefore triggers a re-render of the eval/run/stop buttons.

Instead, make END_INTERRUPT_EXECUTION explicitly (and redundantly) set isRunning in the context to false, and also push an interrupt error into the context ([reference](https://github.com/source-academy/js-slang/blob/13b4e1adbfaed6f1f369e4bd21265867290c3f3b/src/index.ts#L62)).